grapple/types/pages.py

```diff
@@ -117,22 +117,34 @@ class Meta:
         interfaces = (PageInterface,)
 
 
-def get_specific_page(id, slug, token, content_type=None):
+def get_specific_page(id, slug, url, token, content_type=None):
     """
     Get a spcecific page, also get preview if token is passed
     """
     page = None
     try:
+        # Query page based using id, slug, url
+        pages = WagtailPage.objects.live().public()
         if id:
-            page = WagtailPage.objects.live().public().specific().get(pk=id)
+            page = pages.get(id=id)
         elif slug:
-            page = WagtailPage.objects.live().public().specific().get(slug=slug)
+            page = pages.get(slug=slug)
+        elif url:
+            for matching_page in pages.filter(url_path__contains=url):
+                if matching_page.url.strip("/") == url.strip("/"):
```

Comment:
is this the best way of doing it? wouldn't 

```python
pages = pages.filter(url_path=url.lstrip('/'))
if pages.exists():
    page = pages.first()
```

suffice?

Reply:
Yep has been swapped out with:
```
pages = pages.filter(url_path__contains=url.lstrip("/"))
            if pages.exists():
                page = pages.first()
```